geo/src/algorithm/rotate.rs

```diff
@@ -194,7 +194,14 @@ where
 {
     /// Rotate the contained Polygons about their centroids by the given number of degrees
     fn rotate(&self, angle: T) -> Self {
-        MultiPolygon(self.iter().map(|poly| poly.rotate(angle)).collect())
+        match self.centroid() {
+            Some(centroid) => {
+                MultiPolygon(self.iter().map(|poly| poly.rotate_around_point(angle, centroid)).collect())
+            },
+            None => {
```

Comment:
Can you add something like `debug_assert!(self.is_empty())` to document why this could happen?

Reply:
Sorry - this was a bad suggestion. Having no members is only one of a handful of reasons a multipolygon might not have a centroid. Additionally it might have polygons, but *they* might be empty, or their rings otherwise degenerate.

Maybe worth a comment though:
```
// Multipolygon was empty or otherwise degenerate and had no computable centroid
```